src/lib/coda_base/transaction_union_payload.ml

```diff
@@ -229,14 +219,9 @@ type var =
 
 type payload_var = var
 
-let to_hlist ({common; body} : (_, _) User_command_payload.Poly.t) =
-  H_list.[common; body]
-
-let of_hlist : type c v.
-    (unit, c -> v -> unit) H_list.t -> (c, v) User_command_payload.Poly.t =
- fun H_list.[common; body] -> {common; body}
-
 let typ : (var, t) Typ.t =
+  let to_hlist = User_command_payload.Poly.to_hlist in
```

Comment:
could do `let to_hlist,of_hlist = User_command_payload.Poly.(to_hlist,of_list)` to be a little terser